### DISCUSSION TITLE
Caching and refresh token

### DISCUSSION THREAD

**Discussion by Salim-Alsaeh** [marcreichel/igdb-laravel#30] [Q&A] [answered]
Hi, i love your elequent work here, but I have some questions if you can answer them i would be grateful.
- does this package generates new app tokens on its own or it should be done manually?
- how does caching work in this package exactly?

Thank you

**Comment by marcreichel** [comment, marked as answer]
Hi, thank you for your compliment. :)

1. The package retrieves it‘s access token by itself. No manual steps necessary.
2. The package uses your global configured cache driver. Filesystem, Database, etc. It caches the query results (for that specific query) for your configured timespan and then fetches the new results afterwards.

**Reply by Salim-Alsaeh** [reply]
What if the access token is expired?

**Reply by Salim-Alsaeh** [reply]
And do you have any tips to avoid the rate limit ( 4 requests in one second)?
Thanks a million for your response buddy 🙏

**Reply by marcreichel** [reply]
It refreshes the token automatically using your client id and client secret.

**Reply by marcreichel** [reply]
As the package caches your queries by default you should not run into any rate limiting soon. If so, try to talk with the guys from IGDB. They are very responsive and helpful in suiting your needs. :)

**Reply by Salim-Alsaeh** [reply]
Okay, thanks man, I really appreciate it

**Reply by marcreichel** [reply]
No worries, glad to help 😊
